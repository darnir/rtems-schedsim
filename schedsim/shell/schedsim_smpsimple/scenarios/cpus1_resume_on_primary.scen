#
#  Check suspending self, then resuming on primary CPU
#
rtems_init 1
task_create smp1 5
# GOAL: 0x0a010001 is heir and executing on CPU 0
cpus
# suspend running task
task_suspend smp1
# GOAL: 0x09010001 is heir and executing on CPU 0
cpus
# GOAL: 0x09010001 is heir and executing on CPU 0
cpus
# resume suspended task
task_resume smp1
# GOAL: 0x0a010001 is heir and executing on CPU 0
cpus
# GOAL: 0x0a010001 is heir and executing on CPU 0
cpus
