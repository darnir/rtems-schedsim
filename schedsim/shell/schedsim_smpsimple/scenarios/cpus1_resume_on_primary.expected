Script File               : cpus1_resume_on_primary.scen
verbose                   : 1
==> 1: #
==> 2: #  Check suspending self, then resuming on primary CPU
==> 3: #
==> 4: rtems_init 1
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
==> 5: task_create smp1 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  CPU 0: Thread Heir: 0x0a010001 priority=5
  CPU 0: Thread Executing: 0x0a010001 priority=5
==> 6: # GOAL: 0x0a010001 is heir and executing on CPU 0
==> 7: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 8: # suspend running task
==> 9: task_suspend smp1
Suspending task (0x0a010001)
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
==> 10: # GOAL: 0x09010001 is heir and executing on CPU 0
==> 11: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
=== End of Ready Set of Threads
==> 12: # GOAL: 0x09010001 is heir and executing on CPU 0
==> 13: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010001 @255 / 0x09010001 @255          false
=== End of Ready Set of Threads
==> 14: # resume suspended task
==> 15: task_resume smp1
Resuming task (0x0a010001)
  CPU 0: Thread Heir: 0x0a010001 priority=5
  CPU 0: Thread Executing: 0x0a010001 priority=5
==> 16: # GOAL: 0x0a010001 is heir and executing on CPU 0
==> 17: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 18: # GOAL: 0x0a010001 is heir and executing on CPU 0
==> 19: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
